### PR TITLE
python3Packages.parselmouth: init at 0.4.4

### PR DESCRIPTION
**Pull request by DerDennisOP**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+buildPythonPackage rec {
+  pname = "parselmouth";
+  version = "0.4.4";
+  pyproject = true;
+  build-system = [
+    cmake
+    scikit-build
+    setuptools
+    wheel
```
Comment (natsukium) [inline pkgs/development/python-modules/parselmouth/default.nix:24]:
```suggestion
```
We don't need to add `wheel` to inputs.

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    sha256 = "sha256-SbzWk4u1Dt/944N+hiAu/zd49dUXTR2BXs2Ki8fZHiI=";
+  };
+
+  postPatch = ''
+    cp -r --no-preserve=all ${fmt.outPath}/* ./extern/fmt
+  '';
+
+  configurePhase = ''
+    export MAKEFLAGS=-j$NIX_BUILD_CORES
+  '';
```
Comment (natsukium) [inline pkgs/development/python-modules/parselmouth/default.nix:50]:
```suggestion
```
When using cmake, parallel builds are automatically enabled.
https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-enableParallelBuilding

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    setuptools
+    wheel
+  ];
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "YannickJadoul";
+    repo = "Parselmouth";
+    # https://github.com/YannickJadoul/Parselmouth/issues/130
```
Comment (natsukium) [inline pkgs/development/python-modules/parselmouth/default.nix:32]: Could you add a short description of this issue?

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "parselmouth" ];
+
+  meta = {
+    description = "Praat in Python, the Pythonic way";
+    homepage = "https://github.com/YannickJadoul/Parselmouth";
+    changelog = "https://github.com/YannickJadoul/Parselmouth/releases/tag/v${version}";
+    license = lib.licenses.mit;
```
Comment (natsukium) [inline pkgs/development/python-modules/parselmouth/default.nix:68]:
```suggestion
    license = lib.licenses.gpl3Plus;
```
https://github.com/YannickJadoul/Parselmouth/blob/c2cbecc0ce4a0b5d3052cc6c6fbddf4bf133655d/setup.py#L1-L16

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  fmt = fetchFromGitHub {
+    owner = "fmtlib";
+    repo = "fmt";
+    rev = "7bdf0628b1276379886c7f6dda2cef2b3b374f0b";
+    sha256 = "sha256-SbzWk4u1Dt/944N+hiAu/zd49dUXTR2BXs2Ki8fZHiI=";
+  };
+
+  postPatch = ''
+    cp -r --no-preserve=all ${fmt.outPath}/* ./extern/fmt
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/parselmouth/default.nix:44]: Why not use pkgs.fmt?

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+buildPythonPackage rec {
+  pname = "parselmouth";
+  version = "0.4.4";
+  pyproject = true;
+  build-system = [
+    cmake
+    scikit-build
+    setuptools
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/parselmouth/default.nix:24]:
```suggestion
```

File: pkgs/development/python-modules/parselmouth/default.nix [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  postPatch = ''
+    cp -r --no-preserve=all ${fmt.outPath}/* ./extern/fmt
+  '';
+
+  configurePhase = ''
+    export MAKEFLAGS=-j$NIX_BUILD_CORES
+  '';
+
+  dontUseCmakeConfigure = true;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/parselmouth/default.nix:52]:
```suggestion

  build-system = [
    cmake
    scikit-build
    setuptools
  ];

```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval
@ofborg build python312Packages.parselmouth

**Pull request merged**